### PR TITLE
wording clarifications

### PR DESCRIPTION
**Pull request by anikaraghu**
Mostly nits I noticed while reading through that can help make it easier to read

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,84 +16,83 @@ We should be as specific and thorough as possible when defining our style, testi
 
 #### 1. Names of internal functions in a library should not have an underscore prefix.
 
-The style guide states
-
-> Underscore Prefix for Non-external Functions and Variables
-
-One of the motivations for this rule is that it is a helpful visual clue.
+YES:
```
Comment (anikaraghu) [inline README.md:19]: I found it more readable to put the "YES" examples before the "NO" examples - also matches solidity's style guide

File: README.md [hunk 2]
```diff
@@ -16,84 +16,83 @@ We should be as specific and thorough as possible when defining our style, testi
...
+Custom errors should be used wherever possible (an exception could be introducing inconsistencies to existing code bases) - they allow for passing of useful information and are in some cases more gas efficient.
 
 #### 2. Custom error names should be CapWords style.
 
 For example, `InsufficientBalance`.
 
-#### 3. Events
-
-##### A. Events names should be past tense.
+#### 3. Events names should be past tense.
```
Comment (anikaraghu) [inline README.md:59]: I think we were getting to have too many levels which made it hard to follow

Comment (anikaraghu) [reply README.md:59]: we could also instead link some sections out to different READMEs as an alternative

File: README.md [hunk 3]
```diff
@@ -16,84 +16,84 @@ We should be as specific and thorough as possible when defining our style, testi
...
+NO:
 
 ```solidity
 Library._function()
 ```
 
-or
+If a function should never be called from another contract, it should be marked private and its name should have a leading underscore.
+
+#### Justification
```
Comment (wilsoncusack) [inline README.md:33]: We don't have `Justification` elsewhere where we explain reasoning, so I am hesitant to mix up the format. (I used to have something like this but removed for consistency)

File: README.md [hunk 4]
```diff
@@ -16,84 +16,84 @@ We should be as specific and thorough as possible when defining our style, testi
...
+
+We agree that a leading underscore is a useful visual clue, and this is why we oppose using them for internal library functions that can be called from other contracts. Visually, it looks wrong.
 
 ```solidity
 using Library for bytes
 bytes._function()
 ```
 
-Note, we cannot remedy this by insisting on the use public functions. Whether a library functions are internal or external has important implications. From the [Solidity documentation](https://docs.soliditylang.org/en/latest/contracts.html#libraries)
+We could remedy this by insisting on the use public functions. However, developers may prefer internal functions because they are more gas efficient to call, due to how libraries are compiled in Solidity:
```
Comment (wilsoncusack) [inline README.md:46]: imo is confusing to say "we could" remedy. Maybe like "We should not remedy this by ..."

File: README.md (hunk 2 above)
Comment (wilsoncusack) [reply README.md:60]: imo grouping is helpful and separate readmes would require more work, but we have talked about making a doc site. Agree we are pushing a single thing to the limit

**Review by wilsoncusack — COMMENTED** [review]
Thanks for this! There's significant formatting changes mixed in with wording changes here that makes this hard to get in together.

File: README.md (hunk 3 above)
Comment (anikaraghu) [reply README.md:33]: I think exceptions should require justification. This is the only exception we've listed so I would expect future ones to have justification as well, but might not be present right now

File: README.md (hunk 2 above)
Comment (anikaraghu) [reply README.md:60]: hmm I guess it's a bit confusing that some are grouped and some aren't. Should we group all then?

**Review by xenoliss — DISMISSED** [review]

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>
